grails-app/services/org/pih/warehouse/inventory/TransactionEventService.groovy

```diff
@@ -21,8 +20,20 @@ class TransactionEventService implements ApplicationListener<TransactionEvent> {
         Transaction transaction = event?.source
         def transactionId = transaction?.id
         def transactionDate = transaction?.transactionDate
-        def locationId = transaction?.inventory?.warehouse?.id
-        log.info "Refresh inventory snapshot date=$transactionDate, location=$locationId, transaction=$transactionId"
-        RefreshInventorySnapshotJob.triggerNow([location: locationId, forceRefresh: event.forceRefresh])
+        def locationId = event.associatedLocation
+        List productIds = event.associatedProducts
```

Comment:
You don't need to pass this in the event. You can probably just pull this from the transaction. 
```
List productIds = transaction.associatedProducts
```

Reply:
I was afraid of getting LazyInitialization exceptions with pulling it from the transaction in the service. I'll check tomorrow morning if it works that way properly and I refactor it if it will be ok.